app/models/smart_proxy.rb

```diff
@@ -91,7 +89,7 @@ def has_feature?(feature)
   private
 
   def proxy_status
-    @proxy_status ||= ProxyStatus.new(self)
+    @proxy_status ||= ProxyStatus::Base.new(self)
```

Comment:
Just a thought - maybe make this method public, and have it return a hash of features => status objects. Then you can use for example `@smart_proxy.proxy_status[:tftp]` instead of `ProxyStatus::TFTP.new(@smart_proxy)` every time you want to get info from the status object? This method will create a status object for each feature that has one the first time it is called.